guarddog/analyzer/sourcecode/download-executable.yml

```diff
@@ -10,24 +10,33 @@ rules:
       - patterns:
           - pattern-either:
               - pattern: |
-                  $FILE = open("$LOC", ...)
+                  $FILE = open($LOC, ...)
                   ...
                   $FILE.write($REQUEST)
                   ...
                   $CHANGE_PERMISSIONS
               - pattern: |
-                  with open("$LOC", ...) as $FILE:
+                  with open($LOC, ...) as $FILE:
                     ...
                     $FILE.write($REQUEST)
                   ...
                   $CHANGE_PERMISSIONS
+              - pattern: |
+                  open($LOC, ...).write($REQUEST)
```

Comment:
Will this cause a full source->sink match if $REQUEST is modified or changed to a different variable along the way?

Reply:
It depends on the complexity of changes / modifications applied but yes it should match. Do you have anything in mind that we could test?

Reply:
Not particularly. Well, if we ever see malware samples that have a similar pattern that escape this detection, we'll add them as a test case I suppose

Reply:
Exactly